src/js/components/Carousel/Carousel.js

```diff
@@ -1,242 +1,213 @@
 import React, {
   Children,
-  useCallback,
   useContext,
-  useEffect,
   useState,
-  useRef,
+  useEffect,
+  useCallback,
 } from 'react';
-
-import { normalizeColor } from '../../utils';
-import { defaultProps } from '../../default-props';
-import { ThemeContext } from '../../contexts';
-
-import { Box } from '../Box';
-import { Button } from '../Button';
-import { Keyboard } from '../Keyboard';
-import { Stack } from '../Stack';
-
 import { CarouselChild } from './CarouselChild';
+import { CarouselControls } from './CarouselControls';
+import { StyledCarouselContainer } from './StyledCarousel';
 import { CarouselPropTypes } from './propTypes';
+import { Keyboard } from '../Keyboard';
+import { ThemeContext } from '../../contexts';
+import { defaultProps } from '../../default-props';
 
 const Carousel = ({
   activeChild,
-  initialChild,
-  onChild,
-  play,
+  continuous,
   children,
   controls,
+  height,
   fill,
-  onFocus,
-  onBlur,
+  width,
+  initialChild,
+  onChild,
+  play,
   ...rest
 }) => {
+  const containerRef = React.createRef();
   const theme = useContext(ThemeContext) || defaultProps.theme;
-  const [focus, setFocus] = useState();
-  const timerRef = useRef();
-
-  const [indexes, setIndexes] = useState({
-    activeIndex: activeChild !== undefined ? activeChild : initialChild,
+  const numSlides = children.length;
+  const noContainer = !fill && (!height || !width);
+  const animationDuration =
```

Comment:
Since this is included as a dependency for effects, I think we should `useMemo()` to avoid re-running the effects when not needed.